### ISSUE TITLE
2.16.0 run jest got could not find module error

### ISSUE THREAD

**Issue by mydavidzhang** [UI5/webcomponents-react#7964]
### Describe the bug

- Cannot find module '@ui5/webcomponents-react-base/Device' from 'node_modules/@ui5/webcomponents-react/dist/components/AnalyticalTable/hooks/useCanUseVoiceOver.js'
- cannot find module '@ui5/webcomponents-react-base/internal/utils' from 'node_modules/@ui5/webcomponents-react/dist/components/ActionSheet/index.js'
- Cannot find module '@ui5/webcomponents-react-base/internal/hooks' from 'node_modules/@ui5/webcomponents-react/dist/components/AnalyticalTable/scrollbars/VerticalScrollbar.js'

So as a workaround each project need to update jest.config.js

// Workaround for UI5 WebComponents React v2.16.0 bugs
jestConfig.moduleNameMapper = {
  ...jestConfig.moduleNameMapper,
  "@ui5/webcomponents-react-base/Device": "@ui5/webcomponents-react-base/dist/Device", // Cannot find module '@ui5/webcomponents-react-base/Device' from 'node_modules/@ui5/webcomponents-react/dist/components/AnalyticalTable/hooks/useCanUseVoiceOver.js'
  "@ui5/webcomponents-react-base/internal/utils": "@ui5/webcomponents-react-base", // Cannot find module '@ui5/webcomponents-react-base/internal/utils' from 'node_modules/@ui5/webcomponents-react/dist/components/ActionSheet/index.js'
  "@ui5/webcomponents-react-base/internal/hooks": "@ui5/webcomponents-react-base" // Cannot find module '@ui5/webcomponents-react-base/internal/hooks' from 'node_modules/@ui5/webcomponents-react/dist/components/AnalyticalTable/scrollbars/VerticalScrollbar.js'
};

### Reproduction steps

1. Upgrade to "@ui5/webcomponents": "2.16.1" (and others), and "@ui5/webcomponents-react": "2.16.0",
2. npm run test
Get  below errors:

- Cannot find module '@ui5/webcomponents-react-base/Device' from 'node_modules/@ui5/webcomponents-react/dist/components/AnalyticalTable/hooks/useCanUseVoiceOver.js'
- cannot find module '@ui5/webcomponents-react-base/internal/utils' from 'node_modules/@ui5/webcomponents-react/dist/components/ActionSheet/index.js'
- Cannot find module '@ui5/webcomponents-react-base/internal/hooks' from 'node_modules/@ui5/webcomponents-react/dist/components/AnalyticalTable/scrollbars/VerticalScrollbar.js'

### Expected Behaviour

No js error and no workaround should be applied to jest.config.js

### UI5 Web Components for React Version

2.16.0

### UI5 Web Components Version

2.16.1

### Browser

Chrome

### Relevant log output

```shell

```

### Declaration

- [x] I’m not disclosing any internal or sensitive information.

**Comment by mydavidzhang** [comment]
Organization: SAP SF

**Comment by Lukas742** [comment]
Hi @mydavidzhang 

The imports that Jest reports as missing are all correctly defined in the `package.json` [exports map](https://github.com/UI5/webcomponents-react/blob/main/packages/base/package.json) of `@ui5/webcomponents-react-base` ([StackBlitz example](https://stackblitz.com/edit/github-urh54oxf?file=src%2FApp.tsx)). These were introduced in[ v2.15.0](https://github.com/UI5/webcomponents-react/releases/tag/v2.15.0) and the package has been an ES module (`"type": "module"`) for at least more than two years. The previously recommended root barrel import is also still supported.

According to the [Jest ESM documentation](https://jestjs.io/docs/ecmascript-modules#differences-between-native-ecmascript-modules-and-commonjs), Jest supports ESM partially, for example by enabling `extensionsToTreatAsEsm` and running tests in ESM mode (`--experimental-vm-modules`). Did you try that already?

Since we haven’t supported testing with Jest for a while, I’m not fully up to date with the latest Jest and JSDom developments. I don’t see this as a bug, but rather as a limitation of Jest, as package.json export maps are common practice. That said, I can take a closer look if you can provide a reproducible example of the issue.
